macros/Compat.ejs

```diff
@@ -1,66 +1,399 @@
+          }
+        } else {
+          let noteIndex = compatNotes.indexOf(supportData.notes);
+          supportInfo = `${supportInfo} <sup><a href="#compatNote_${noteIndex+1}">${noteIndex+1}</a></sup>`;
+        }
+      }
+
+      // there is a flag and it needs a note, too
+      if (supportData.flag) {
+        if (compatNotes) {
+          let flagNote = generateFlagsNote(supportData, browser);
+          let noteIndex = compatNotes.indexOf(flagNote);
+          supportInfo = `${supportInfo} <sup><a href="#compatNote_${noteIndex+1}">${noteIndex+1}</a></sup>`;
+        }
+    }
+
+  }
+  return supportInfo;
 }
 
+/*
+Generate an aggregate table that summarises the support status for the
+individual features under the data it's passed.
+Each row in the table will contain the "basic_support" data for a single feature.
+The first cell in each the row will contain a link to the feature's page.
+
+`json_identifier` is an `identifier` JSON object.
+*/
+function writeAggregateTable(json_identifier, environment, browserNames) {
+  let output = writeTableHead(environment, browserNames);
+  output += "<tbody>";
+  // add a row for each feature
+  for (let featureName of Object.keys(json_identifier).sort()) {
+    output += "<tr>";
+    let featureData = json_identifier[featureName]["__compat"];
+    // the first cell is the name of the feature
+    output += `<td><code>${featureName}</code></td>`;
+    // then one cell for each browser
+    for (let browserNameKey of Object.keys(browserNames)) {
+      let supportData = featureData["basic_support"]["support"][browserNameKey];
+      let supportInfo = '';
+      // if supportData is an array, there are multiple support statements
+      if (Array.isArray(supportData)) {
+        for (let entry of supportData) {
+          supportInfo += getSupportInfo(entry, browserNameKey);
+          if (supportInfo) {supportInfo += "<br/>"; };
+        }
+      } else { // there is just one support statement
+        supportInfo = getSupportInfo(supportData, browserNameKey);
+      }
+      output += `<td>${supportInfo}</td>`;
+    }
+    output += "</tr>";
+  }
+  output += "</tbody></table></div>";
+  return output;
+}
+
+/*
+Iterate into all "support" objects, and all browsers under them,
+and collect all notes in an array, without duplicates.
+
+`json_subfeature_set` is a `subfeature_set` JSON object.
+*/
+function collectCompatNotes(json_subfeature_set) {
+
+  function pushNotes(supportEntry, browserName) {
+    if (supportEntry.hasOwnProperty("notes")) {
+      let notes = supportEntry["notes"];
+      if (Array.isArray(notes)) {
+        for (let note of notes) {
+          if (notesArray.indexOf(note) === -1) {
+            notesArray.push(note);
+          }
+        }
+      } else {
+        if (notesArray.indexOf(notes) === -1) {
+          notesArray.push(notes);
+        }
+      }
+    } else if (supportEntry.hasOwnProperty("flag")) {
```

Comment:
The `else` means that if a supportEntry has both notes and a flag, then the flagNote will not be added here.